src/code-editor/interfaces.ts

```diff
@@ -105,8 +112,11 @@ export interface CodeEditorProps extends BaseComponentProps, FormFieldControlPro
   onEditorContentResize?: NonCancelableEventHandler<CodeEditorProps.ResizeDetail>;
 }
 
+type LiteralUnion<LiteralType, BaseType extends string> = LiteralType | (BaseType & { _?: never });
+type BuiltInLanguage = typeof AceModes[number]['value'];
+
 export namespace CodeEditorProps {
-  export type Language = typeof AceModes[number]['value'];
+  export type Language = LiteralUnion<BuiltInLanguage, string>;
```

Comment:
Since the builtin languages are strings too, isn't this line the same as saying `export type Language = string`?